### ISSUE TITLE
F***** CPU affinity get me mad!

### ISSUE THREAD

**Issue by ArSd-g** [RainbowMiner/RainbowMiner#1562]
Tired of trying to deal with f*** usless cpu affinity.....
Explain for exemple;
Panthera Algo: with auto-affinity (24 threads) : 0x555555555555 = 5,7kh/s
If i run miner outside RBM without "cpu-affinity=0x555555555555" parameter = 8,5kh/s etc for a lot of algos
I tried to set "0" as affinity in config.txt but i get:
"WARNING: Parameter CPUMiningAffinity (config.txt) is empty or contains errors. Falling back to 0x555555555555"
Can you please find a way to get rid of this f*** nightmare...KICKING OFF THIS USELESS PARAMETER, xmrig, srb-miner and more just "msr tweak" automatically....just selecting nbrs of threads would be ok...i can't guess what affinity they use...

**Comment by RainbowMiner** [comment]
Try 
```
  "CPUAffinity": "0xFFFFFFFFFFF5",
  "CPUThreads": "46",
```
.. that will help.

**Comment by ArSd-g** [comment]
It worked! as always....thank you!
Correct me if i'm wrong; i can set this affinity (0xFFFFFFFFFFF5 ; 46T ; exclude T1 and T3) in RBM config.txt and in the same time use only 24T on a specify miner using this affinity?

If you want a great "alternative pool side" idea to implement; i use [https://github.com/MoneroOcean/xmrig](url) Version as a basis to compare and set in RBM...#Autobenchmark

**Comment by RainbowMiner** [comment]
> Correct me if i'm wrong; i can set this affinity (0xFFFFFFFFFFF5 ; 46T ; exclude T1 and T3) in RBM config.txt and in the same time use only 24T on a specify miner using this affinity?

Yes, that is very well possible. Regarding xmrig, it's a bit different - RainbowMiner uses the xmrig's build-in default and "logic AND"s it with the config.txt affinity setup. So to say, for heavy algorithms (like the CryptonightHeavyXhv for example) it will always use optimum affinity/number of threads. If you want to adjust affinity/threads for xmrig, you will have to go to folder `.\Bin\ANY-Xmrig` and find the right `config_... .json` file and edit it.
